### PR TITLE
Add support for using ranges for RGB

### PR DESCRIPTION
**Pull request by Erovia**
This patch adds support for controlling continuous ranges of RGB LEDs.
Helper functions for split boards are also available.
New functions are also present in the documentation.

### CODE REVIEW

File: quantum/rgblight.c [hunk 1]
```diff
@@ -562,6 +562,42 @@ void rgblight_sethsv_at(uint16_t hue, uint8_t sat, uint8_t val, uint8_t index) {
...
+void rgblight_setrgb_range(uint8_t r, uint8_t g, uint8_t b, uint8_t start, uint8_t end) {
+  if (!rgblight_config.enable || start < 0 || start >= end || end > RGBLED_NUM) { return; }
+
+  for (uint8_t i = start; i < end; i++) {
+    led[i].r = r;
+    led[i].g = g;
+    led[i].b = b;
+  }
+  rgblight_set();
+  _delay_ms(1);
```
Comment (drashna) [inline quantum/rgblight.c]: Please change this to the hardware agnostic version of this function:
```suggestion
  wait_ms(1);
```

**Comment by drashna** [comment]
The `*_master` and *_slave` functions should probably be wrapped by ifdef's, so that they don't do anything on non-split keyboards. 

Also.... I don't think they will work on i2c split boards, because of how the code is set, but I could be wrong.

**Comment by Erovia** [comment]
> The `*_master` and *_slave` functions should probably be wrapped by ifdef's, so that they don't do anything on non-split keyboards.
> 
> Also.... I don't think they will work on i2c split boards, because of how the code is set, but I could be wrong.

I thought about it, but wasn't sure what condition to use. "SPLIT_KEYBOARD" is not yet present in every split's rules.mk, is it?
Anyway, they only control the first/second half of the LEDs, so I don't think they cause much trouble.
(or maybe I'm just naive :D)

I could only test the code on a Let's  Split rev2, which afaik uses i2c and it works fine.

File: quantum/rgblight.c (hunk 1 above)
Comment (Erovia) [reply quantum/rgblight.c]: Thanks for pointing this out.
It's indeed more elegant this way.

**Comment by Erovia** [comment]
@drashna, does this PR needs anyting else?

**Pull request merged**